### PR TITLE
substrate 2.0.0-rc4 version upgrade

### PR DESCRIPTION
**Pull request by iorveth**
This PR aims to upgrade content directory to substrate version 2.0.0-rc4

Closes: https://github.com/Joystream/joystream/issues/1054

To minimize merge conflict potential, this PR assumes #1064 will be merged first.

### CODE REVIEW

**Comment by bedeho** [comment]
I think I will just leave this review to @shamil-gadelshin

**Review by bedeho — APPROVED** [review]

File: runtime-modules/content-directory/Cargo.toml [hunk 1]
```diff
@@ -1,40 +1,34 @@
...
-# https://users.rust-lang.org/t/failure-derive-compilation-error/39062
-quote = '<=1.0.2'
+codec = { package = 'parity-scale-codec', version = '1.3.1', default-features = false, features = ['derive'] }
+rstd = { package = 'sp-std', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = '00768a1f21a579c478fe5d4f51e1fa71f7db9fd4'}
+sp-runtime = { package = 'sp-runtime', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = '00768a1f21a579c478fe5d4f51e1fa71f7db9fd4'}
+frame-support = { package = 'frame-support', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = '00768a1f21a579c478fe5d4f51e1fa71f7db9fd4'}
+system = { package = 'frame-system', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = '00768a1f21a579c478fe5d4f51e1fa71f7db9fd4'}
+sp-arithmetic = { package = 'sp-arithmetic', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = '00768a1f21a579c478fe5d4f51e1fa71f7db9fd4'}
```
Comment (shamil-gadelshin) [inline runtime-modules/content-directory/Cargo.toml]: Let's have a common format for dependencies: single line based similar to parity's format.

File: runtime-modules/content-directory/Cargo.toml [hunk 2]
```diff
@@ -1,40 +1,34 @@
...
 optional = true
 version = '1.0.101'
 
 [dev-dependencies]
-runtime-io = { package = 'sr-io', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'c37bb08535c49a12320af7facfd555ce05cce2e8'}
-primitives = { package = 'substrate-primitives', git = 'https://github.com/paritytech/substrate.git', rev = 'c37bb08535c49a12320af7facfd555ce05cce2e8'}
+sp-io = { package = 'sp-io', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = '00768a1f21a579c478fe5d4f51e1fa71f7db9fd4'}
+sp-core = { package = 'sp-core', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = '00768a1f21a579c478fe5d4f51e1fa71f7db9fd4'}
 
 [features]
```
Comment (shamil-gadelshin) [inline runtime-modules/content-directory/Cargo.toml:20]: Could you sort feature-definitions similar to dependencies order: it's easier to check them.

File: runtime-modules/content-directory/Cargo.toml [hunk 3]
```diff
@@ -1,40 +1,34 @@
...
 default = ['std']
 std = [
 	'serde',
 	'codec/std',
 	'rstd/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/content-directory/Cargo.toml]: Could you rename rstd to sp-std - similar to the new format in the Substrate?

File: runtime-modules/content-directory/Cargo.toml [hunk 4]
```diff
@@ -1,40 +1,34 @@
 [package]
 name = 'substrate-content-directory-module'
```
Comment (shamil-gadelshin) [inline runtime-modules/content-directory/Cargo.toml]: Please rename it to 'pallet-content-directory'.

File: runtime-modules/content-directory/src/lib.rs [hunk 5]
```diff
@@ -5,47 +5,54 @@
...
 pub use serde::{Deserialize, Serialize};
 
-pub use errors::*;
-pub use helpers::*;
-pub use operations::*;
-pub use permissions::*;
-pub use schema::*;
-
 /// Type, used in diffrent numeric constraints representations
 type MaxNumber = u32;
```
Comment (shamil-gadelshin) [inline runtime-modules/content-directory/src/lib.rs:49]: If it's not too far out your current scope could you perform actual error conversion to the Substrate decl_error! macro?

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
We don't have a strict rule for that but I recommend to comment all public methods or type definitions (for example in the storage definition): it could be done with temporary enabling this code line: `#![warn(missing_docs)]` .

Also, I like that for such a complex module it's very clean and understandable. Good work!

File: runtime-modules/content-directory/Cargo.toml (hunk 4 above)
Comment (iorveth) [reply runtime-modules/content-directory/Cargo.toml]: Done here https://github.com/Joystream/joystream/pull/1072/commits/0e4cfb15c7a0d866f490a5ccae9fcd472b9df295

File: runtime-modules/content-directory/Cargo.toml (hunk 3 above)
Comment (iorveth) [reply runtime-modules/content-directory/Cargo.toml]: Done https://github.com/Joystream/joystream/pull/1072/commits/0e4cfb15c7a0d866f490a5ccae9fcd472b9df295

File: runtime-modules/content-directory/Cargo.toml (hunk 2 above)
Comment (iorveth) [reply runtime-modules/content-directory/Cargo.toml:20]: Good point, https://github.com/Joystream/joystream/pull/1072/commits/0e4cfb15c7a0d866f490a5ccae9fcd472b9df295

File: runtime-modules/content-directory/Cargo.toml (hunk 1 above)
Comment (iorveth) [reply runtime-modules/content-directory/Cargo.toml]: Done https://github.com/Joystream/joystream/pull/1072/commits/0e4cfb15c7a0d866f490a5ccae9fcd472b9df295

**Comment by iorveth** [comment]
_We don't have a strict rule for that but I recommend to comment all public methods or type definitions (for example in the storage definition): it could be done with temporary enabling this code line: #![warn(missing_docs)] ._

Thanks for pointing that feature. Added missing comments for the all public functions, traits and structure fields. 
https://github.com/Joystream/joystream/pull/1072/commits/b570a9a91758f578937d3f87316621ef3c003e99

File: runtime-modules/content-directory/src/lib.rs (hunk 5 above)
Comment (iorveth) [reply runtime-modules/content-directory/src/lib.rs:49]: Done here https://github.com/Joystream/joystream/pull/1072/commits/0ed9ecd9b5958aeeaa032451e1579d35cda22d35,
https://github.com/Joystream/joystream/pull/1072/commits/65cdabbfec7ce1803a4f7cc7768f9f491b73baf5

**Review by shamil-gadelshin — APPROVED** [review]
LGTM. Nice work on error conversion!

**Pull request merged**